Update queries_controller.rb

adrianlfns opened the pull request:
Please review the pull request below. I had to add this specifically for the measure 165. That measure have multiple population sets so the statement  qc = CQM::QualityReport.where('measure_id' => measure.id, 'effective_date' => options[:effective_date],'start_date' => options[:start_date], "sub_id" => params[:sub_id], "filters.providers" => {'$in': params[:providers]}).first  does not find any value.

I added the statement below in order to prevent an exception when the qc.status is called.

  if qc.nil?
             qc = CQM::QualityReport.where('measure_id' => measure.id, 'effective_date' => options[:effective_date],'start_date' => options[:start_date], "filters.providers" => {'$in': params[:providers]}).first 
            end

adrianlfns committed:
Update queries_controller.rb

Please review the pull request below. I had to add this specifically for the measure 165. That measure have multiple population sets so the statement 
qc = CQM::QualityReport.where('measure_id' => measure.id, 'effective_date' => options[:effective_date],'start_date' => options[:start_date], "sub_id" => params[:sub_id], "filters.providers" => {'$in': params[:providers]}).first 
does not find any value.

I added the statement below in order to prevent an exception when the qc.status is called.

  if qc.nil?
             qc = CQM::QualityReport.where('measure_id' => measure.id, 'effective_date' => options[:effective_date],'start_date' => options[:start_date], "filters.providers" => {'$in': params[:providers]}).first 
            end

petercyli reviewed (commented):
I didn't get an error when testing measure 165.  Not sure whether it is related to other changes that removed a potential for qc to be nil.  Will ask another developer to review this request.  Thanks.

petercyli committed:
Update queries_controller.rb

It makes no sense to search another qc, because the error causing qc.status problem was deleting the cache before failing (No cache = qc is nil = no matter if you search without status).
